forge-ai/src/main/java/forge/ai/GameState.java

```diff
@@ -1245,6 +1249,16 @@ private CardCollectionView processCardsForZone(final String[] data, final Player
                     c.setBackSide(true);
                 } else if (info.startsWith("Flipped")) {
                     c.setState(CardStateName.Flipped, true);
+                } else if (info.startsWith("MeldedWith")) {
+                    String meldCardName = info.substring(info.indexOf(':') + 1).replace("^", ",");
+                    Card meldTarget;
+                    PaperCard pc = StaticData.instance().getCommonCards().getCard(meldCardName);
+                    if (pc == null) {
+                        System.err.println("ERROR: Tried to create a non-existent card named " + meldCardName + " (as a MeldedWith card) when loading game state!");
+                        continue;
+                    }
+                    meldTarget = Card.fromPaperCard(pc, c.getOwner());
+                    c.setMeldedWith(meldTarget);
                 } else if (info.startsWith("Meld")) {
```

Comment:
looks good

As a minor improvement maybe fold the "|Meld" part into it too, imo there's no case where you would want to only use one of it alone

Reply:
Yep, good call :) Updated this in a followup PR.